diff.py

```diff
@@ -1895,11 +1914,23 @@ def _jump_table_entries_count(self, raw_lines: List[str], line_no: int) -> int:
         for i in reversed(range(line_no)):
             cmp_match = re.search(ARM32_COMPARE_IMM_PATTERN, raw_lines[i])
             if cmp_match:
-                value = immediate_to_int(cmp_match.group(2))
+                value = int(cmp_match.group(2).lstrip("#"), 0)
                 if value > 0:
                     return value + 1
         return 0
 
+    def pre_process(
+        self, mnemonic: str, args: str, next_row: Optional[str], comment: Optional[str]
+    ) -> Tuple[str, str]:
+        arg_parts = args.split()
+        # Normalize instructions that omit the destination register.
+        if len(arg_parts) == 2 and any(
+            [insn in mnemonic for insn in ARM32_DEST_OPTIONAL]
+        ):
```

Comment:
```suggestion
        if len(arg_parts) == 2 and any(ins in mnemonic for ins in ARM32_DEST_OPTIONAL):
```
making use of generators.

A bit silly to split on whitespace instead of `", "`, but works fine in this case.